### PR TITLE
chore: Add Quiltflower decompiler support and greatly improve workspace/development documentation

### PR DESCRIPTION
**Pull request by kristofbolyai**
Quiltflower overall improves decompiled code quality, but it also includes the code comments of the source code, and show useful information like `@Override` annotations.

Example of the `Quiltflower` improvements (compared to `FernFlower`):

FernFlower:
image

QuiltFlower: 
image

FernFlower:
image

QuiltFlower:
image

FernFlower:
image

QuiltFlower:
image

### CODE REVIEW

**Comment by kristofbolyai** [comment]
Before merging, I would like to get a review from both @magicus and @Incompleteusern (mostly about the docs changes).

File: README.md
Comment (Incompleteusern) [inline README.md]: is a

Comment (kristofbolyai) [reply README.md]: fixed, thanks

**Review by magicus — APPROVED** [review]

**Review by Incompleteusern — APPROVED** [review]

**Comment by magicus** [comment]
I was eager to try this out, but I think the readme update was incorrect. The gradle target seems to be named `genSourcesWithQuiltflower`, unless I'm messing things up.

**Comment by kristofbolyai** [comment]
> I was eager to try this out, but I think the readme update was incorrect. The gradle target seems to be named `genSourcesWithQuiltflower`, unless I'm messing things up.

It is! Sorry about that!

**Pull request merged**
